ext/intl/calendar/gregoriancalendar_methods.cpp

```diff
@@ -208,6 +220,82 @@ U_CFUNC PHP_METHOD(IntlGregorianCalendar, __construct)
 	}
 }
 
+U_CFUNC PHP_METHOD(IntlGregorianCalendar, createFromDate)
+{
+	zend_long year, month, day;
+	UErrorCode status = U_ZERO_ERROR;
+	zend_error_handling error_handling;
+	Calendar_object *co;
+	GregorianCalendar *gcal;
+
+	intl_error_reset(NULL);
+
+	if (zend_parse_parameters(ZEND_NUM_ARGS(), "lll", &year, &month, &day) == FAILURE) {
+		RETURN_THROWS();
+	}
+
+	ZEND_VALUE_ERROR_OUT_OF_BOUND_VALUE(year, 1);
+	ZEND_VALUE_ERROR_OUT_OF_BOUND_VALUE(month, 2);
+	ZEND_VALUE_ERROR_OUT_OF_BOUND_VALUE(day, 3);
+
+	zend_replace_error_handling(EH_THROW, IntlException_ce_ptr, &error_handling);
+
+	gcal = new GregorianCalendar((int32_t) year, (int32_t) month, (int32_t) day, status);
+	if (!set_gregorian_calendar_time_zone(gcal, status)) {
+		delete gcal;
+		goto cleanup;
+	}
```

Comment:
I'd rather have a boolean parameter passed to ``set_gregorian_calendar_time_zone()`` to know if it should throw then replacing the error handling.

Reply:
Hmm, although this would mean that I cannot use `intl_error_set()` anymore and have to have ifs everywhere. + I still have to call `intl_error_set_code()` in order to keep consistency with the original code... So I'm not really sure if getting rid of replacing the error handler is a good idea

Reply:
I suppose this can be done as a follow-up. Probably deprecating the warning mode would also make sense...